### ISSUE TITLE
Convert existing array-based mask to a regionmask object

### ISSUE THREAD

**Issue by rjp23** [mathause/regionmask#137]
I don't think this is currently possible but it'd be extremely useful. 

I have an existing numpy array that I use to mask my data for certain regions, e.g. if value is 1, region is A, if value is 2, region is B, etc.

Being able to create a regionmask object from this array (e.g. where array == x) would be very useful.

**Comment by mathause** [comment]
Yes, this is currently not possible. There is a draft/ proof of concept of this in #87 but I haven't followed up on this since.

**Comment by rjp23** [comment]
Thanks, yep that's the same sort of issue. A lot of these existing scientific regions already come as "masks" rather than polygons or boundaries and converting them to regionmask would be really nice.

**Comment by mathause** [comment]
I had another look at this and found the following form.  Assuming `cond` is a boolean array (i.e. we would iterate over each region separately) this would look as follows:


```python

def to_multipolygon(cond):
    
    assert np.issubdtype(cond.dtype, bool)
    
    transform = regionmask.core.mask._transform_from_latlon(cond.lon, cond.lat)

    res = rasterio.features.shapes(
        cond.astype(np.int16), cond, transform=transform, connectivity=8
    )

    poly = list()
    for coords, val in res:
        shell, *holes = coords["coordinates"]
        p = shapely.Polygon(shell, holes)

        poly.append(p)

    return shapely.MultiPolygon(poly)
```

The one thing that I did not find documented is that the coords have the form `shell, *holes`.
